docs/src/utils/demoState.ts

```diff
@@ -0,0 +1,13 @@
+/*
+
+We are using a Map as a lightweight solution to persist component demo state across NextJS re-renders without the complexity of state management (e.g., Redux). 
```

Comment:
Add a comment on why we need to persist demo state with a link to the original github issue.

Reply:
Added